rfd/0109-opsgenie-plugin.md

```diff
@@ -0,0 +1,176 @@
+# Opsgenie Integration RFD
+## Required approvers
+
+Engineering: @r0mant, @marcoandredinis, @hugoShaka
+Product: @klizhentas, @xinding33
+Security: @reedloden, @jentfoo
+
+## What
+
+This RFD proposes a plugin that allows Teleport to integrate with Opsgenie, allowing access requests to show up as alerts in Opsgenie. This plugin will differ from existing plugins by being part of the Teleport binary directly. 
+Doing so by creating a new Teleport service 'plugin_service' with the opsgenie plugin (and any future ones) being configurable resources. This will allow the plugins to be run either by themselves or alongside other Teleport services.
+
+## Scope
+
+### Phase 1
+
+The plugin will be run as a standalone application similar to the existing Pagerduty plugin differing only in that it will be part of the Teleport binary.
+
+### Phase 2
+
+The plugin will follow the 'hosted' operations model and will at that point be possible to run as part of the Auth server directly or as a standalone application.
```

Comment:
I think these phases are no longer accurate. I would update the scope to the following:

- The plugin will be able to run standalone or alongside other services as part of the new `plugin_service`.
- The plugin will support auto-approval similar to PagerDuty plugin.